front/lib/api/assistant/conversation.ts

```diff
@@ -1434,6 +1448,18 @@ export async function* retryAgentMessage(
     if (!messageRow || !messageRow.agentMessage) {
       return null;
     }
+    const newerMessage = await Message.findOne({
+      where: {
+        rank: messageRow.rank,
+        conversationId: conversation.id,
+        version: {
+          [Op.gt]: messageRow.version,
+        },
+      },
+    });
+    if (newerMessage) {
+      throw new Error("Unexpected: Message to retry is not the latest version");
```

Comment:
If you throw you won't fix anything you want to emit an error event and return from the function early no?

Reply:
Throw would still drop the stream right?